src/options.cpp

```diff
@@ -364,6 +394,12 @@ void cOpt::setNext()
 
         sSet = vItems[iNext];
 
+    } else if (sType == "string_input") {
+        int iMenuTextLength = sMenuText.length();
+        sSet = string_input_popup("", (iMaxLength > 80) ? 80 : ((iMaxLength < iMenuTextLength) ? iMenuTextLength : iMaxLength+1),
```

Comment:
`string_input_popup` returns an empty string when the user cancels the input (via escape key). This should probably result in the setting not being changed.

There is also the more difficult problem that canceling here will still set `bChangedSomething` to `true` in `show_options`, even through nothing has been changed. User enters the option menu, selects the new option, presses enter (input box appears), presses escape to cancel, presses escape again to leave the option screen. The game now asks whether to save the changes (what changes?). But this can probably be tolerated.